### ISSUE TITLE
[Feature request] "Open URL with..." option in Download File dialog

### ISSUE THREAD

**Issue by magiblot** [QupZilla/qupzilla#2585]
Hi there,

Some players such as mpv support playing multimedia directly in streaming mode. If I get to a download dialog but I don't want to spend time downloading the file, I will copy the URL, open a terminal and run the application with the URL as input parameter: I would find it very useful to be able to do this directly from the download dialog. It could also serve as a workaround for those installations of qupzilla lacking audio/video codecs.

I'm not sure of how useful this could be for other filetypes.

Thanks.

PS.: I'm enclosing two screenshots which describe what I said.
![screenshot_20180201_124943](https://user-images.githubusercontent.com/20713561/35675283-b9c7d9ee-0747-11e8-9f3b-8062706c3b25.png)
![screenshot_20180201_125033](https://user-images.githubusercontent.com/20713561/35675284-b9e5864c-0747-11e8-9497-827b238bec78.png)

**Comment by cranes-bill** [comment]
> It could also serve as a workaround for those installations of qupzilla lacking audio/video codecs.

QupZilla does not lacking anything!!! You have to read more about **proprietary codecs**!
By the way, did you ever try the [**Videoner**](https://github.com/QupZilla/qupzilla-plugins/tree/master/plugins/Videoner) plugin?

**Comment by magiblot** [comment]
> QupZilla does not lacking anything!!! You have to read more about proprietary codecs!
Excuse me, I used the wrong words.
> By the way, did you ever try the Videoner plugin?
I didn't know about this plugin! It might be what I'm looking for.

**Comment by cranes-bill** [comment]
> I didn't know about this plugin! It might be what I'm looking for.

Maybe not exactly, but you might find it useful.

Also, as workaround you can create a simple shell script which can pass the media files to mpv or save it to the disk, and then set this script as download manager in Qupzilla.

**Comment by potholiday** [comment]
@magiblot  I was like you this will solve your problem https://github.com/QupZilla/qupzilla-plugins/issues/75

**Comment by magiblot** [comment]
> @magiblot I was like you this may solve your problem QupZilla/qupzilla-plugins#75

Videoner is very useful for YouTube, but I'm not having problems with this page at the moment. It doesn't help with download links.
> Also, as workaround you can create a simple shell script which can pass the media files to mpv or save it to the disk, and then set this script as download manager in Qupzilla.

Yes, it would take some time but this seems the simpliest way to attain the behaviour I'm looking for.

**Comment by potholiday** [comment]
@cranes-bill I was trying to recreate the problem of @magiblot. I wrote a simple 'mpv`script to stream those mp3's
```
#!/bin/sh
mpv $1
```
and added this executable in both the `general` and `media elements` tabs in videoner. The problem is `download` button in the site https://my-free-mp3.net/ redirects to a different site and the `continue` button in the second site dynammically generates a download link. Because the download link is dynamically generated we can only get the orginal download link only in the download box `Copy download link` where third mouse key is disabled.

![image](https://user-images.githubusercontent.com/12268027/35716048-3cea8f24-07fc-11e8-90dc-b75ff0342a1a.png)

**Comment by cranes-bill** [comment]
That what i meant is quite different. Just create script which detect the file type and if it's media file pass it to **mpv**, otherwise use **wget** for example, then set it here:
![capture du 2018-02-02 09 07 39](https://user-images.githubusercontent.com/5161278/35721163-790913c0-07f9-11e8-9053-5f95a1873f85.png)

**Comment by potholiday** [comment]
I didnt know that we can fix an external download manger in qupzilla. This is a nice feature but for now I am gonna stick with the native download manager. Can I ask what is the `Arguments` box here means. I have seen this box in both the download manger and videoner plugin. Is there any documentation about it?

**Comment by cranes-bill** [comment]
You might set binary file as executable, so if it need rguments, you can set it here.
For example, if if you set `/bin/wget` as executable, you can set `--continue --timeout=60 %d ` and so on as arguments

**Comment by magiblot** [comment]
> Also, as workaround you can create a simple shell script which can pass the media files to mpv or save it to the disk, and then set this script as download manager in Qupzilla.

I have been thinking about this. I came to this idea:

In addition to the "External download manager", the user could be allowed to set "Custom commands". A custom command would consist of a executable, a list of arguments and a name, so that the user can identify them.

In the Download File dialog there would be a third option named "Run Command" where the user could choose one of their preconfigured custom commands or create a new one.

This would be an efficient and friendly way of helping the user in these cases where they need to execute the same command for each file they follow the link of, which could be several times.

Excuse me for asking you to implement any of these. I don't have the knowledge yet to program this on my own either as a workaround or as a pull request.

**Comment by cranes-bill** [comment]
By the way, with simple modification, Videoner plugin can do the similar thing.
![capture du 2018-02-05 18 25 04](https://user-images.githubusercontent.com/5161278/35816793-4ff65b2a-0aa4-11e8-96c4-80efd4c5070f.png)

**Comment by magiblot** [comment]
Indeed, that is simpler than my whole idea. Maybe the easiest way to have this working with all links (including the dinamically generated ones, as @potholiday noticed) would be making the right-button work on the link in the Download File dialog.

**Comment by dachnigel971** [comment]
Why not try the site: https://freemp3downloads.online. Free and easy use.

**Comment by magiblot** [comment]
Please, no. Not only the purpose of my request wasn't downloading music (it was just an example), but the site you linked will generate transcodes of YouTube videos (which are already transcodes on their own). No way.
